CONTRIBUTING.md

```diff
@@ -47,44 +48,47 @@ the expected behavior that doesn't occur.
 
 To suggest an enhancement, open an
 [issue](https://github.com/nginx/unit/issues/new) on GitHub with the label
-`enhancement`.  Please do this before implementing a new feature to discuss the
-feature first.
+`enhancement`.  Please do this before implementing a new feature to discuss
+the feature first.
 
 
 ### Open a Pull Request
 
-Before submitting a PR, please read the NGINX Unit code guidelines to know more
-about coding conventions and benchmarks.  Fork the repo, create a branch, and
-submit a PR when your changes are tested and ready for review.  Again, if you'd
-like to implement a new feature, please consider creating a feature request
-issue first to start a discussion about the feature.
+Before submitting a PR, please read the NGINX Unit code guidelines to know
+more about coding conventions and benchmarks.  Fork the repo, create a branch,
+and submit a PR when your changes are tested and ready for review.  Again, if
+you'd like to implement a new feature, please consider creating a feature
+request issue first to start a discussion about the feature.
 
 
 ## Git Style Guide
 
-- Keep a clean, concise and meaningful `git commit` history on your branch,
-  rebasing locally and squashing before submitting a PR
+- Create atomic commits.  A commit should do just one thing, I.e you
+  shouldn't mix whitespace changes with actual code changes.  Do the
+  whitespace changes in a separate commit first.
 
-- For any user-visible changes, updates, and bugfixes, add a note to
-  `docs/changes.xml` under the section for the upcoming release, using `<change
-  type="feature">` for new functionality, `<change type="change">` for changed
-  behavior, and `<change type="bugfix">` for bug fixes.
+  Ideally you should rebase locally and force push new commits up.
 
-- In the subject line, use the past tense ("Added feature", not "Add feature");
-  also, use past tense to describe past scenarios, and present tense for
-  current behavior
+- In the subject line, use the imperative mood.  I.e write the subject like
+  you're giving git a command, e.g "Free memory before exiting". Do _not_
```

Comment:
E.g. is also an abbreviation, thus both periods are needed. Punctuation mark in this case goes inside the quotation marks, see https://www.grammarly.com/blog/quotation-marks/

Highlighting the _not_ is a bit harsh in my opinion, we can be friendlier than that. 🙂 
```suggestion
  you're giving git a command, e.g. "Free memory before exiting." Do not
```

Reply:
This is an example of a commit subject and the '.' is _not_ part of the commit subject.

Reply:
Good point, I retract this changeset, thanks!